core/l10n_init.js

```diff
@@ -61,6 +61,8 @@ define('core/l10n_init',
     }
 
     function _transformLocale(locale) {
+        // Use default locale "en-US" if given locale is undefined (BUG 1138558)
+        locale = locale || 'en-US';
```

Comment:
i checked the situation again and the error `TypeError: locale is undefined in l10n_init.js` only occures if the user has no browser language set, even before the language file gets injected. but i can open the production marketplace @ marketplace.firefox.com without problems, even without a default browser language. so i can't tell you if the changes in the l10n_init.js are needed or not. The error shouldn't be affected by the missing language file.